--- ydpdict-0.52/ydpdict.conf.example~	Sat Nov  9 02:46:12 2002
+++ ydpdict-0.52/ydpdict.conf.example	Sat Nov  9 02:46:12 2002
@@ -2,7 +2,7 @@
 # dosowej przy kernelu 2.0 jest z³ym pomys³em, poniewa¿ dostêp bêdzie trwa³
 # kilkakrotnie d³u¿ej ni¿ do partycji ext2.
 
-Path /usr/local/share/ydpdict
+Path /usr/share/ydpdict
 
 # ¦cie¿ka, gdzie znajduje siê podmontowany CD-ROM ze s³ownikiem. U¿ywane do
 # odtwarzania próbek d¼wiêkowych.
