--- /home/grzegol/rpm/SOURCES/aaa/ydpdict-0.51/ydpdict.c	Sun Oct 17 16:08:28 1999
+++ ydpdict-0.51/ydpdict.c	Sat May 25 21:00:53 2002
@@ -61,9 +60,10 @@
 void sigterm();
 void redrawdef();
 void preparewins();
+void changedict(char *name);
 
 /* do dzie³a panowie... */
-void main(int argc, char **argv) {
+int main(int argc, char **argv) {
   int ch;
 
   /* na mój sygna³... */
@@ -137,11 +137,22 @@
 hmm... tak w³a¶ciwie, to nie wiem co mia³bym tutaj wrzuciæ. wystarczy \
 powiedzieæ, ¿e klawisz TAB prze³±cza okna, kursorami oraz PgUp i PgDn \
 mo¿na przewijaæ zarówno listê s³ów, jak i definicjê s³owa. klawisz F2 \
-powoduje odtworzenie próbki d¼wiêkowej.\
+powoduje odtworzenie próbki d¼wiêkowej, a F3 i F4 zmieniaj± odpowiednio \
+s³ownik na angielsko-polski i polsko-angielski.\
 }\\par\\pard{\
 kontakt z autorem: {\\b wojtekka@irc.pl}\
 }"));
	break;
+      case KEY_F(3):
+        changedict("en-pl");
+				defline=0;
+				defupd=1;
+	break;
+			case KEY_F(4):
+				changedict("pl-en");
+				defline=0;
+				defupd=1;
+	break;
       case KEY_UP:
         if (defmark) {
    if (defline > 0) defline--;
@@ -211,6 +221,7 @@
   } while(ch != 27);
   
   showerror(NULL);
+	return 0;
 }
 
 /* zakoñczenie killem */
@@ -502,4 +513,31 @@
 void sigsegv() {
   signal(SIGSEGV, SIG_IGN);
   showerror(_("Naruszenie ochrony pamiêci (skontaktuj siê z autorem programu)"));
+}
+
+/* Zmienia s³ownik */
+void changedict(char *name)
+{
+	int pl=0, i;
+	closedict();
+	if(!strncasecmp(name, "pl-en", 5)) pl=1;
+
+  if (!opendict(filespath, pl ? DEFINDEX_PA : DEFINDEX_AP, pl ? DEFDICT_PA : DEFDICT_AP)) switch (ydperror) {
+    case YDP_OUTOFMEMORY:
+      showerror(_("Brak pamiêci."));
+    case YDP_CANTOPENIDX:
+      showerror(_("Nie mo¿na otworzyæ pliku indeksowego."));
+    case YDP_CANTOPENDEF:
+      showerror(_("Nie mo¿na otworzyæ pliku z definicjami."));
+    case YDP_INVALIDFILE:
+      showerror(_("B³±d podczas czytania plików."));
+  }
+
+	defline=0;
+	defupd=1;
+	menu=0;
+	pos=0;
+
+	for(i=0; i<strlen(input); i++)
+		input[i]=0;
 }
