diff -urN aaa/ydpdict-0.51/ydpdict.c ydpdict-0.51/ydpdict.c
--- aaa/ydpdict-0.51/ydpdict.c	Sun Oct 17 16:08:28 1999
+++ ydpdict-0.51/ydpdict.c	Sat May 25 19:28:58 2002
@@ -131,7 +131,7 @@
         if (playsample(pos + menu) < 1) beep();
 	break;
       case KEY_F(1):
-        def = _("\
+        def = strdup(_("\
 {\\b ydpdict-" VERSION "\\line\\cf1(c) 1999, 1998 by wojtek kaniewski}\
 \\par\\pard{\
 hmm... tak w³a¶ciwie, to nie wiem co mia³bym tutaj wrzuciæ. wystarczy \
@@ -140,7 +140,7 @@
 powoduje odtworzenie próbki d¼wiêkowej.\
 }\\par\\pard{\
 kontakt z autorem: {\\b wojtekka@irc.pl}\
-}");
+}"));
 	break;
       case KEY_UP:
         if (defmark) {
